File: routes/typescript-to-php/certification/artifacts/holdout/source-runtime/source_harness.ts
import { clamp as _elmosHarnessSubject } from "./clamp.js";
function _elmosHarnessFP64(value: number): string {
  const bytes = new ArrayBuffer(8);
  const view = new DataView(bytes);
  view.setFloat64(0, value, false);
  return view.getBigUint64(0, false).toString(16).padStart(16, "0");
}
function _elmosHarnessHexUTF8(value: string): string {
  return Array.from(new TextEncoder().encode(value), byte => byte.toString(16).padStart(2, "0")).join("");
}
const actual0 = _elmosHarnessSubject(20, 10);
const expected0 = 10;
if (!Number.isFinite(actual0) || !Object.is(actual0, expected0)) throw new Error("case 0");
console.log("ELMOS_OBSERVATION\t0\tfp64-hex\t" + _elmosHarnessFP64(actual0));
const actual1 = _elmosHarnessSubject(-2, 10);
const expected1 = 0;
if (!Number.isFinite(actual1) || !Object.is(actual1, expected1)) throw new Error("case 1");
console.log("ELMOS_OBSERVATION\t1\tfp64-hex\t" + _elmosHarnessFP64(actual1));
const actual2 = _elmosHarnessSubject(7, 10);
const expected2 = 7;
if (!Number.isFinite(actual2) || !Object.is(actual2, expected2)) throw new Error("case 2");
console.log("ELMOS_OBSERVATION\t2\tfp64-hex\t" + _elmosHarnessFP64(actual2));

File: routes/python-to-cpp/certification/artifacts/development/source-runtime/source_harness.py
import json
import math
import struct
import _elmos_source_module
expected_0 = 120
actual_0 = _elmos_source_module.calculate(100, 20)
assert type(actual_0) is int
assert actual_0 == expected_0
print("ELMOS_OBSERVATION\tjson\t" + json.dumps({"case_id": 0, "value": actual_0}, sort_keys=True, separators=(",", ":")))
expected_1 = 0
actual_1 = _elmos_source_module.calculate(-1, 5)
assert type(actual_1) is int
assert actual_1 == expected_1
print("ELMOS_OBSERVATION\tjson\t" + json.dumps({"case_id": 1, "value": actual_1}, sort_keys=True, separators=(",", ":")))
expected_2 = 5
actual_2 = _elmos_source_module.calculate(7, -2)
assert type(actual_2) is int
assert actual_2 == expected_2
print("ELMOS_OBSERVATION\tjson\t" + json.dumps({"case_id": 2, "value": actual_2}, sort_keys=True, separators=(",", ":")))
